import "goby/middleware/protobuf/app_config.proto";
import "goby/zeromq/protobuf/interprocess_config.proto";
import "goby/middleware/protobuf/udp_config.proto";
import "goby/middleware/protobuf/io.proto";

import "dccl/option_extensions.proto";

message NetSimUDPConfig
{
    option (dccl.msg).unit_system = "si";

    optional goby.middleware.protobuf.AppConfig app = 1;
    optional goby.zeromq.protobuf.InterProcessPortalConfig interprocess = 2;

    required goby.middleware.protobuf.UDPOneToManyConfig udp = 20;

    message ModemEndPoint
    {
        required uint32 modem_tcp_port = 1;
        required uint32 modem_id = 2;
        required goby.middleware.protobuf.UDPEndPoint endpoint = 3;
    }
    repeated ModemEndPoint modem = 30;

    optional bool send_all_as_broadcast = 40 [default = true];

    message RangeToBernoulliProbability
    {
        required double range = 1;
        required double success_probability = 2;
    }

    repeated RangeToBernoulliProbability range_to_packet_success = 50;
    optional double sound_speed = 51
        [default = 1500, (dccl.field).units.derived_dimensions = "velocity"];
}
